### ISSUE TITLE
Upgrading to EMQX 5.0.23 from 5.0.22 appears to lose previously configured settings and break.

### ISSUE THREAD

**Issue by ruifung** [emqx/emqx#10474]
### What happened?

When upgrading from EMQX 5.0.22 to 5.0.23, certain settings do not appear to be preserved.
For example. I had authorization rules and authentication accounts set up via admin interface.
After upgrading to 5.0.23, it lists no authentication sources and no authorization sources. Authentication does still appear to work (i.e. previously created username/password pairs still work)
however, authorization appears to be broken as I've observed services on my home network are unable to listen to messages.

### What did you expect to happen?

For existing settings and configuration to be carried across with no issues. Especially since according to the version number, this should be a patch release.

### How can we reproduce it (as minimally and precisely as possible)?

1. (From a pre-5.0.23 version), create username (internal DB), and authorization (file)
2. Upgrade to 5.0.23
3. Observe admin UI no longer shows them, and authorization rules seem to be wierd.

### Anything else we need to know?

For now, I've downgraded back to 5.0.22. Being able to control the lights is kind of important. For now, I do not have logs from 5.0.23.

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
# paste output here
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
NOT APPLICABLE. Running on Talos Linux v1.3.3 / Kubernetes, does not contain console.
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Log files

<details>

</details>

**Comment by savonarola** [comment]
Hello!
* How many nodes does your setup have?
* What is the deployment and upgrade method used for Kubernetes?
* Did the authentication and authorization return back to normal after the downgrade back to 5.0.23?
* What mechanism is used to provide persistence for EMQX nodes?

**Comment by nicandris** [comment]
yup had the same issue (docker). i couldnt login on the dashboard and no device could connect. I had to change all references in the config files of message_in to messages, bytes_in to bytes etc (check changelog)

but now i keep getting errors while navigating menus in the dashboard eg

`500 INTERNAL_ERROR:error, {config_not_found,[<<"mqtt">>]}, [{emqx_config,do_get,2,[{file,"emqx_config.erl"},{line,671}]},{emqx_mgmt_api_configs,get_config_with_default,1,[{file,"emqx_mgmt_api_configs.erl"},{line,305}]},{emqx_mgmt_api_configs,'-get_zones/0-fun-0-',2,[{file,"emqx_mgmt_api_configs.erl"},{line,298}]},{lists,foldl,3,[{file,"lists.erl"},{line,1267}]},{emqx_mgmt_api_configs,global_zone_configs,3,[{file,"emqx_mgmt_api_configs.erl"},{line,210}]},{minirest_handler,apply_callback,3,[{file,"minirest_handler.erl"},{line,111}]},{minirest_handler,handle,2,[{file,"minirest_handler.erl"},{line,44}]},{minirest_handler,init,2,[{file,"minirest_handler.erl"},{line,27}]},{cowboy_handler,execute,2,[{file,"cowboy_handler.erl"},{line,41}]},{cowboy_stream_h,execute,3,[{file,"cowboy_stream_h.erl"},{line,318}]},{cowboy_stream_h,request_process,3,[{file,"cowboy_stream_h.erl"},{line,302}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]`

**Comment by ruifung** [comment]
> Hello!
> 
> * How many nodes does your setup have?
> * What is the deployment and upgrade method used for Kubernetes?
> * Did the authentication and authorization return back to normal after the downgrade back to 5.0.23?
> * What mechanism is used to provide persistence for EMQX nodes?
1. 3 main, 3 replicas
2. Emqx-operator
3. Yes
4. PVC, built in db

**Comment by zhongwencool** [comment]
> yup had the same issue (docker). i couldnt login on the dashboard and no device could connect. I had to change all references in the config files of message_in to messages, bytes_in to bytes etc (check changelog)

HI @nicandris Could you provide the complete configuration file, please? If you're a single node, it may not be the same issue.

**Comment by dsilletti** [comment]
Upgrading to 5.0.23 requires a modified cluster-override.conf, some different settings are causing emqx to crash:

e.g. in the listeners settings:

- bytes_in
- message_in
- capacity

and so on. Please check the logs and fix all the settings reported
